### PR TITLE
Issue #2894035: Setup the Register config entity

### PR DESCRIPTION
**Pull request by EvanSchisler**

### CODE REVIEW

**Comment by smmccabe** [comment]
Yo, what is up with this title! terrible!  Seriously though, can you match it to an issue plz.

File: config/schema/register.schema.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+commerce_pos.register.*:
```
Comment (bojanz) [inline config/schema/register.schema.yml]: This file is misnamed, it needs to have the module name. "commerce_pos.schema.yml"

File: src/Entity/Register.php [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+ *   },
+ *   config_prefix = "register",
+ *   admin_permission = "administer site configuration",
+ *   entity_keys = {
+ *     "id" = "id",
+ *     "label" = "label",
+ *     "uuid" = "uuid"
+ *   },
+ *   links = {
+ *     "canonical" = "/admin/structure/pos/register/register/{register}",
```
Comment (bojanz) [inline src/Entity/Register.php]: 1) Why aren't the urls under /admin/commerce/? Either /admin/commerce/pos or /admin/commerce/config/pos?
2) Why do registers have a canonical url, they're not meant to be viewed?

File: src/Entity/Register.php [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+namespace Drupal\commerce_pos\Entity;
+
+use Drupal\Core\Config\Entity\ConfigEntityBase;
+
+/**
+ * Defines the Register entity.
+ *
+ * @ConfigEntityType(
+ *   id = "register",
```
Comment (bojanz) [inline src/Entity/Register.php]: Entity type IDs need to be prefixed with the module name. "commerce_pos_register".

File: src/RegisterHtmlRouteProvider.php [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+use Drupal\Core\Entity\Routing\AdminHtmlRouteProvider;
+use Symfony\Component\Routing\Route;
+
+/**
+ * Provides routes for Register entities.
+ *
+ * @see Drupal\Core\Entity\Routing\AdminHtmlRouteProvider
+ * @see Drupal\Core\Entity\Routing\DefaultHtmlRouteProvider
+ */
+class RegisterHtmlRouteProvider extends AdminHtmlRouteProvider {
```
Comment (bojanz) [inline src/RegisterHtmlRouteProvider.php]: This is just Drupal Console boilerplate, you don't need this class at all.

File: src/Form/RegisterDeleteForm.php [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+namespace Drupal\commerce_pos\Form;
+
+use Drupal\Core\Entity\EntityConfirmFormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Url;
+
+/**
+ * Builds the form to delete Register entities.
+ */
+class RegisterDeleteForm extends EntityConfirmFormBase {
```
Comment (bojanz) [inline src/Form/RegisterDeleteForm.php]: You don't need this class at all. Just specify Drupal\Core\Entity\EntityDeleteForm in the entity annotation.

File: src/Entity/Register.php [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+ *     "form" = {
+ *       "add" = "Drupal\commerce_pos\Form\RegisterForm",
+ *       "edit" = "Drupal\commerce_pos\Form\RegisterForm",
+ *       "delete" = "Drupal\commerce_pos\Form\RegisterDeleteForm"
+ *     },
+ *     "route_provider" = {
+ *       "html" = "Drupal\commerce_pos\RegisterHtmlRouteProvider",
+ *     },
+ *   },
+ *   config_prefix = "register",
```
Comment (bojanz) [inline src/Entity/Register.php]: Once you update the entity type ID, update the config prefix as well.

File: src/Entity/Register.php [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+ *   admin_permission = "administer site configuration",
```
Comment (bojanz) [inline src/Entity/Register.php]: POS has its own permission, use it.

File: src/Form/RegisterForm.php [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+  public function form(array $form, FormStateInterface $form_state) {
+    $form = parent::form($form, $form_state);
+
+    $register = $this->entity;
+    $form['label'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Label'),
+      '#maxlength' => 255,
+      '#default_value' => $register->label(),
+      '#description' => $this->t("Label for the Register."),
```
Comment (bojanz) [inline src/Form/RegisterForm.php]: The #description here adds nothing to the UI, remove it.

File: src/RegisterFloatInterface.php [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+ *
+ * The amount of money that is in the till at the beginning of each day.
+ */
+interface RegisterFloatInterface {
+
+  /**
+   * Add a float to the register
+   *
+   * @param \Drupal\commerce_pos\Entity\RegisterInterface $register
+   * The register.
```
Comment (bojanz) [inline src/RegisterFloatInterface.php]: Each @param description needs a two-space indent.

**Pull request merged**